File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int toBinary(int n){
    int cnt = 0;
    while(n != 0){
        if(n%2 == 1) cnt++;
        n/=2;
    }
    return cnt;
}

int solution(int n) {
    int a = toBinary(n);
    
    while(1){
        n++;
        if(a == toBinary(n)) break;
    }
    
    return n;
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        String s = Integer.toBinaryString(n);
        for(char c : s.toCharArray()){
            if(c == '1') answer++;
        }
        
        int next = n + 1;
        while(true) {
            int cnt = 0;
            String temp = Integer.toBinaryString(next);
            
            for(char c : temp.toCharArray()){
                if(c == '1') cnt++;
            }
            
            if(answer == cnt) {
                return next;
            }
            next++;
        }
    }
}